modules/@apostrophecms/image/index.js

```diff
@@ -122,6 +122,19 @@ module.exports = {
       }
     }
   },
+  handlers(self) {
+    return {
+      beforeUpdate: {
+        // Ensure the crop fields are updated on publishing an image
+        async autoCropRelations(req, piece, options) {
+          if (piece.aposMode !== 'published') {
```

Comment:
Hmm. Does the logic wind up adjusting references from the drafts too? (Images are autopublished but do have draft and published modes under the hood)

Reply:
My thoughts - "relations" should be valid published version of a document. Even if the original document is a draft, it should reference a published doc (image piece in this case). 

We update all "mode" versions of the referenced document. E.g. a page using the image as a widget will update cropped data for "draft", "publised" and "previous" if crop data is available in the document.

Reply:
OK. As long as a user who has set up a crop in a draft page that's never been published, let's say, will not find a broken image on their page when they use "replace" on that image then I'm fine.

Reply:
This is the case. Once an image is "published", our handler will search for all documents that reference it, no matter what mode are they.